jfr-daemon/src/main/java/com/newrelic/jfr/daemon/JFRDaemon.java

```diff
@@ -38,11 +39,22 @@ public static void main(String[] args) {
       var config = buildConfig();
       var mBeanServerConnection = new MBeanServerConnector(config).getConnection();
       var readinessCheck = new AtomicBoolean(false);
+      var eventConverterReference = new AtomicReference<EventConverter>();
 
-      var entityGuid = new RemoteEntityGuid(mBeanServerConnection).queryFromJmx();
+      RemoteEntityGuidCheck.builder()
+          .mbeanServerConnection(mBeanServerConnection)
+          .onComplete(
+              guid -> {
+                var attr = new JFRCommonAttributes(config).build(guid);
+                var eventConverter = buildEventConverter(attr);
+                eventConverterReference.set(eventConverter);
+                readinessCheck.set(true);
+              })
+          .build()
+          .start();
       readinessCheck.set(true);
```

Comment:
Am I reading this wrong, or are we setting the readinessCheck to true twice here?

Reply:
Nope, good catch!  Will fix.